Dockerfile

```diff
@@ -45,8 +44,9 @@ ENV EDITOR nano
 RUN curl --silent --show-error https://getcomposer.org/installer | \
         php -- --install-dir=/usr/local/bin --filename=composer
 
-# Install Node.js 10.1.0
-RUN npm install -g n && n 10.1.0
+# Install Node.js 10.x
+RUN curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash - && \
+    sudo apt-get install -y nodejs
```

Comment:
I think we can do this more cleanly. Ran into the issue with `cs50/sandbox`, and I think installing both `nodejs` _and_ `npm` fixes, without resorting to cURL:

https://github.com/cs50/sandbox/blob/master/cs50.sh#L28-L29

Reply:
I think I tried that and didn't work.

Reply:
Mind trying again? Does seem to have fixed `cs50/sandbox`.

Reply:
```
$ cat cli/Dockerfile
FROM cs50/baseimage
USER root
ARG DEBIAN_FRONTEND=noninteractive

# Expose ports (just like Cloud9)
EXPOSE 8080 8081 8082

# Install packages
RUN apt-get update && \
    apt-get install -y \
        apt-file \
        bash-completion \
        bc \
        bsdtar \
        cmake \
        dos2unix \
        dnsutils `# for nslookup` \
        exiftool \
        gettext \
        golang-go \
        imagemagick \
        info \
        man \
        nano \
        mysql-client \
        openjdk-11-jdk-headless \
        openjdk-11-jre-headless \
        npm \
        nodejs \
        php \
        rpm \
        ruby \
        ruby-dev `# Avoid "can't find header files for ruby" for gem` \
        s3cmd \
        sudo \
        telnet \
        traceroute \
        tree \
        vim \
        wget \
        whois && \
    apt-file update
ENV EDITOR nano

# Install Composer
RUN curl --silent --show-error https://getcomposer.org/installer | \
        php -- --install-dir=/usr/local/bin --filename=composer

# Install Node.js packages
RUN npm install -g grunt http-server

# Install fpm, asciidoctor
# https://github.com/asciidoctor/jekyll-asciidoc/issues/135#issuecomment-241948040
# https://github.com/asciidoctor/jekyll-asciidoc#development
RUN gem install \
    asciidoctor \
    bundler \
    fpm \
    jekyll-asciidoc \
    pygments.rb

## Install CS50 packages
RUN apt-get update && \
    apt-get install -y \
        libcs50-java \
        php-cs50

# Install Python packages
RUN pip3 install \
    awsebcli \
    awscli `# must come after awsebcli to ensure supported version` \
    help50 \
    render50 \
    submit50 \
    virtualenv

# Install hub
# https://hub.github.com/
RUN wget -P /tmp https://github.com/github/hub/releases/download/v2.4.0/hub-linux-amd64-2.4.0.tgz && \
    tar xvf /tmp/hub-linux-amd64-2.4.0.tgz -C /tmp && \
    /tmp/hub-linux-amd64-2.4.0/install && \
    rm -rf /tmp/hub-linux-amd64-2.4.0*

# Copy files to image
RUN wget --directory-prefix /etc/profile.d/ https://raw.githubusercontent.com/git/git/master/contrib/completion/git-prompt.sh
COPY ./etc/profile.d/cli.sh /etc/profile.d/
COPY ./etc/vim/vimrc.local /etc/vim/
RUN echo "This is CS50 CLI." > /etc/motd

# Add user to sudoers 
RUN echo "\n# CS50 CLI" >> /etc/sudoers
RUN echo "ubuntu ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
RUN echo "Defaults umask_override" >> /etc/sudoers
RUN sed -e "s|^Defaults\tsecure_path=.*|Defaults\t!secure_path|" -i /etc/sudoers
USER ubuntu
```
```
$ docker build -t cs50/ide:online online
...
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./compat -I./include -I./include -g -O2 -Wall -fno-strict-aliasing -pthread -MT bufferevent_openssl.lo -MD -MP -MF .deps/bufferevent_openssl.Tpo -c bufferevent_openssl.c  -fPIC -DPIC -o .libs/bufferevent_openssl.o
bufferevent_openssl.c: In function ‘bio_bufferevent_new’:
bufferevent_openssl.c:106:3: error: dereferencing pointer to incomplete type ‘BIO {aka struct bio_st}’
  b->init = 0;
   ^~
bufferevent_openssl.c: At top level:
bufferevent_openssl.c:228:1: error: variable ‘methods_bufferevent’ has initializer but incomplete type
 static BIO_METHOD methods_bufferevent = {
 ^~~~~~
bufferevent_openssl.c:79:27: warning: excess elements in struct initializer
 #define BIO_TYPE_LIBEVENT 57
                           ^
bufferevent_openssl.c:229:2: note: in expansion of macro ‘BIO_TYPE_LIBEVENT’
  BIO_TYPE_LIBEVENT, "bufferevent",
  ^~~~~~~~~~~~~~~~~
bufferevent_openssl.c:79:27: note: (near initialization for ‘methods_bufferevent’)
 #define BIO_TYPE_LIBEVENT 57
                           ^
bufferevent_openssl.c:229:2: note: in expansion of macro ‘BIO_TYPE_LIBEVENT’
  BIO_TYPE_LIBEVENT, "bufferevent",
  ^~~~~~~~~~~~~~~~~
bufferevent_openssl.c:229:21: warning: excess elements in struct initializer
  BIO_TYPE_LIBEVENT, "bufferevent",
                     ^~~~~~~~~~~~~
bufferevent_openssl.c:229:21: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:230:2: warning: excess elements in struct initializer
  bio_bufferevent_write,
  ^~~~~~~~~~~~~~~~~~~~~
bufferevent_openssl.c:230:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:231:2: warning: excess elements in struct initializer
  bio_bufferevent_read,
  ^~~~~~~~~~~~~~~~~~~~
bufferevent_openssl.c:231:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:232:2: warning: excess elements in struct initializer
  bio_bufferevent_puts,
  ^~~~~~~~~~~~~~~~~~~~
bufferevent_openssl.c:232:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:233:2: warning: excess elements in struct initializer
  NULL /* bio_bufferevent_gets */,
  ^~~~
bufferevent_openssl.c:233:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:234:2: warning: excess elements in struct initializer
  bio_bufferevent_ctrl,
  ^~~~~~~~~~~~~~~~~~~~
bufferevent_openssl.c:234:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:235:2: warning: excess elements in struct initializer
  bio_bufferevent_new,
  ^~~~~~~~~~~~~~~~~~~
bufferevent_openssl.c:235:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:236:2: warning: excess elements in struct initializer
  bio_bufferevent_free,
  ^~~~~~~~~~~~~~~~~~~~
bufferevent_openssl.c:236:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:237:2: warning: excess elements in struct initializer
  NULL /* callback_ctrl */,
  ^~~~
bufferevent_openssl.c:237:2: note: (near initialization for ‘methods_bufferevent’)
bufferevent_openssl.c:228:19: error: storage size of ‘methods_bufferevent’ isn’t known
 static BIO_METHOD methods_bufferevent = {
                   ^~~~~~~~~~~~~~~~~~~
Makefile:793: recipe for target 'bufferevent_openssl.lo' failed
make[2]: *** [bufferevent_openssl.lo] Error 1
make[2]: Leaving directory '/opt/c9/libevent-2.0.21-stable'
Makefile:857: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/opt/c9/libevent-2.0.21-stable'
Makefile:592: recipe for target 'all' failed
make: *** [all] Error 2

real    0m25.248s
user    0m16.862s
sys     0m3.552s
The command '/bin/sh -c curl --silent --location https://raw.githubusercontent.com/c9/install/master/install.sh > /tmp/c9install &&     chmod a+x /tmp/c9install &&     sudo mkdir /opt/c9 &&     sudo chown --recursive ubuntu:ubuntu /opt/c9 &&     sed --in-place 's#C9_DIR=.*#C9_DIR=/opt/c9#' /tmp/c9install &&     /tmp/c9install' returned a non-zero code: 2
```